plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/delete/RowPredicate.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.delete;
+
+import io.trino.spi.Page;
+
+import static java.util.Objects.requireNonNull;
+
+public interface RowPredicate
+{
+    boolean test(Page page, int position);
```

Comment:
Should the testing be batched?
(it will surely be easier once the "and" structure is flattened)

cc @raunaqmorarka @sopel39

Reply:
Most likely. Are there multiple implementations or `RowPredicate`?

Reply:
Yes, 3 (equalities, position, `and`)